### PR TITLE
Update envoy config samples

### LINKED ISSUE #394: refresh example configs / docs
The example configs are slightly out of date based on new features that we now run in prod at Lyft. Refresh the example docs and make the documentation about them a bit more thorough.

### PR DESCRIPTION
**Pull request by ccaraman**
Fixes #394 

Add sample for 
* RDS
* CDS
* Outlier Detection
* Remote Address rate limiting in Envoy Front Proxy template

### CODE REVIEW

**Comment by ccaraman** [comment]
@lyft/network-team

File: configs/envoy_service_to_service.template.json [hunk 1]
```diff
@@ -300,6 +341,17 @@
       "refresh_delay_ms": 30000
     },
 
+    "cds": {
+      "cluster": {
+        "name": "cds",
+        "connect_timeout_ms": 250,
+        "type": "strict_dns",
+        "lb_type": "round_robin",
+        "features": "http2",
```
Comment (mattklein123) [inline configs/envoy_service_to_service.template.json]: I would probably del `features: http2` from CDS and RDS that you added for simplicity.

File: configs/envoy_service_to_service.template.json [hunk 2]
```diff
@@ -156,6 +156,47 @@
         ]
       }
     }]
+  },
+  {
```
Comment (junr03) [inline configs/envoy_service_to_service.template.json]: Leave a comment saying that this listener has a dynamic route table instead of a static one.

Comment (ccaraman) [inline configs/envoy_service_to_service.template.json]: Added a small comment in the template and added a note to the reference configuration docs.

**Pull request merged**